.pipelines/npm/npm-conformance-tests.yaml

```diff
@@ -1,270 +1,449 @@
+
+            # Enable Microsoft.ContainerService/AKSWindows2022Preview
+            az feature register --namespace Microsoft.ContainerService --name AKSWindows2022Preview
+            az provider register -n Microsoft.ContainerService
+
+            export CLUSTER_NAME=$(RESOURCE_GROUP)-$(PROFILE)
+
+            echo "Creating resource group named $CLUSTER_NAME"
+            az group create --name $CLUSTER_NAME -l $(LOCATION) -o table
+
+            echo "Creating resource group named $CLUSTER_NAME"
+            az aks create \
+                --resource-group $CLUSTER_NAME \
+                --name $CLUSTER_NAME \
+                --generate-ssh-keys \
+                --windows-admin-username e2eadmin \
+                --windows-admin-password alpha@numeric!password2 \
+                --network-plugin azure \
+                --vm-set-type VirtualMachineScaleSets \
+                --node-vm-size Standard_D4s_v3 \
+                --kubernetes-version v1.23 \
+                --node-count 1
+
+            # don't schedule anything on the linux system pool
+            echo "Updating $CLUSTER_NAME to not schedule anything on linux pool..."
+            az aks nodepool update \
+              --cluster-name $CLUSTER_NAME \
+              -g $CLUSTER_NAME \
+              -n nodepool1 \
+              --node-taints CriticalAddonsOnly=true:NoSchedule
+
+            echo "Adding Windows nodepool to $CLUSTER_NAME"
+            az aks nodepool add \
+                --resource-group $CLUSTER_NAME \
+                --cluster-name $CLUSTER_NAME \
+                --name awin22 \
+                --os-type Windows \
+                --os-sku Windows2022 \
+                --node-vm-size Standard_D4s_v3 \
+                --node-count 1
+
+            echo "Getting credentials to $CLUSTER_NAME"
+            az aks get-credentials -g $CLUSTER_NAME -n $CLUSTER_NAME --overwrite-existing --file ./kubeconfig
+            mkdir -p ~/.kube/
+            cp ./kubeconfig ~/.kube/config
+
+      - task: AzureCLI@2
+        displayName: "Deploy NPM to Test Cluster"
+        inputs:
+          azureSubscription: $(BUILD_VALIDATIONS_SERVICE_CONNECTION)
+          scriptType: "bash"
+          scriptLocation: "inlineScript"
+          inlineScript: |
+
+            export CLUSTER_NAME=$(RESOURCE_GROUP)-$(PROFILE)
+
+            curl -LO https://dl.k8s.io/release/v1.23.0/bin/linux/amd64/kubectl
+            chmod +x kubectl
+
+            # deploy azure-npm
+            ./kubectl --kubeconfig=./kubeconfig apply -f https://raw.githubusercontent.com/Azure/azure-container-networking/master/npm/examples/windows/azure-npm.yaml
+
+            # swap azure-npm image with one built during run
+            ./kubectl --kubeconfig=./kubeconfig set image daemonset/azure-npm -n kube-system azure-npm=$IMAGE_REGISTRY/azure-npm:windows-amd64-$(TAG)
+
+            # swap NPM profile with one specified as parameter
+            ./kubectl --kubeconfig=./kubeconfig apply -f 
```

Comment:
not applying any file here. Should probably do the following steps:
1. apply both linux and windows yaml
2. swap profile
3. restart both win and linux ds